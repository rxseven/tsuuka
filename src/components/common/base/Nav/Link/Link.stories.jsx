import React from 'react';
import { storiesOf } from '@storybook/react';

import Link from './index';

const title = 'Components/Common/Base/Nav/Link';
const props = { to: '/' };

storiesOf(title, module)
  .add('default', () => <Link {...props}>Nav Link</Link>)
  .add(
    'light',
    () => (
      <Link light {...props}>
        Nav Link
      </Link>
    ),
    {
      notes: 'Link text is white, use with dark background for readability.'
    }
  );
