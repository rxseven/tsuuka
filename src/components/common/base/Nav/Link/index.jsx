import PropTypes from 'prop-types';
import React from 'react';
import { NavLink } from 'react-router-dom';
import styled from 'styled-components';

const Frame = styled(NavLink)`
  :hover {
    text-decoration: none;
  }

  &.${props => props.activeClassName} {
    cursor: default;

    &,
    :hover {
      color: ${props => (props['data-light'] ? '#fff' : '#999')};
    }
  }
`;

const propTypes = {
  children: PropTypes.node.isRequired,
  light: PropTypes.bool
};

const defaultProps = { light: false };

function Link({ children, light, ...props }) {
  return (
    <Frame {...props} activeClassName="active" data-light={light}>
      {children}
    </Frame>
  );
}

Link.propTypes = propTypes;
Link.defaultProps = defaultProps;

export default Link;
