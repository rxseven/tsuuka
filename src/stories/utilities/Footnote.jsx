import React from 'react';
import styled from 'styled-components';

const Frame = styled.div`
  bottom: 2rem;
  font-size: 14px;
  position: fixed;
`;

function Footnote({ children }) {
  return <Frame>{children}</Frame>;
}

export default Footnote;
