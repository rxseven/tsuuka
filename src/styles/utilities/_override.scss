// Reset button's default styles
button:focus {
  box-shadow: none !important;
  outline: 0 !important;
}
