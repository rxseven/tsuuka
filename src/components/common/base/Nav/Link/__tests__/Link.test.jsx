import { factory } from 'tests/utilities';
import Link from '../index';

// Arrange
const source = {
  children: 'Link Text',
  to: '/'
};
const input = { ...source };

// Setup
function setup(props) {
  return factory(Link, source, props);
}

// Test suites
describe('<Nav.Link />', () => {
  it('should render without crashing', () => {
    setup();
  });

  it('should render passed children correctly', () => {
    const expected = { content: input.children };
    const { component } = setup();

    expect(component).toHaveTextContent(expected.content);
  });
});
