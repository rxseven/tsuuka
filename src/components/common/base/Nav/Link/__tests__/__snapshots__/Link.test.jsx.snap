// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Nav.Link /> Snapshot tests should render correctly 1`] = `
<a
  aria-current="page"
  className="sc-bdVaJa cVXdtY active"
  data-light={false}
  href="/"
  onClick={[Function]}
  style={Object {}}
>
  Link Text
</a>
`;
